mk/build.mk

```diff
@@ -1,7 +1,19 @@
 BUILD_INFO_GIT_TAG ?= $(shell git describe --tags 2>/dev/null || echo unknown)
 BUILD_INFO_GIT_COMMIT ?= $(shell git rev-parse HEAD 2>/dev/null || echo unknown)
 BUILD_INFO_BUILD_DATE ?= $(shell date -u +"%Y-%m-%dT%H:%M:%SZ" || echo unknown)
-BUILD_INFO_VERSION ?= $(shell prefix=$$(echo $(BUILD_INFO_GIT_TAG) | cut -c 1); if [ "$${prefix}" = "v" ]; then echo $(BUILD_INFO_GIT_TAG) | cut -c 2- ; else echo $(BUILD_INFO_GIT_TAG) ; fi)
+
+# Kuma version is built as follows:
+# 1) If a git tag is present on the current commit, then the version is a git tag
+# 2) If the branch starts with "release" (like "release-1.3"), then the version is $(last git tag)-$(short_git_hash) (for example: 1.4.1-450174242)
```

Comment:
I have to :+1:, according to semver, `1.4.2-<hash>` is a prerelease version of `1.4.2`. I'm not sure I like thinking of pre release versions that aren't explicit prerelease version (like `beta`, etc) and postrelease semantics might fit better a lot of the time, but if we care about versioning in a non Kuma specific way it may be the only option.

Reply:
Ok, I changed the script to increment the patch